client/src/entrypoints/admin/workflow-action.js

```diff
@@ -121,4 +124,53 @@ function ActivateWorkflowActionsForEditView(formSelector) {
     );
   });
 }
+/** @deprecated RemovedInWagtail70 - Remove global.ActivateWorkflowActionsForEditView usage  */
 window.ActivateWorkflowActionsForEditView = ActivateWorkflowActionsForEditView;
+
+const currentScript = document.currentScript;
+const activateTarget = currentScript.dataset.activate;
+const cancellationUrl = currentScript.dataset.confirmCancellationUrl;
+
+document.addEventListener('DOMContentLoaded', () => {
+  if (activateTarget === 'dashboard') {
+    ActivateWorkflowActionsForDashboard();
+  } else if (activateTarget === 'editor') {
+    ActivateWorkflowActionsForEditView('[data-edit-form]');
+  }
+
+  if (cancellationUrl) {
+    /* Make user confirm before publishing the object if it will cancel an ongoing workflow */
+    let cancellationConfirmed = false;
+    $('[name=action-publish]').click((e) => {
+      if (!cancellationConfirmed) {
+        e.stopImmediatePropagation();
+        e.preventDefault();
+        window.ModalWorkflow({
+          url: cancellationUrl,
+          onload: {
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            confirm(modal, jsonData) {
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              $('[data-confirm-cancellation]', modal.body).click((event) => {
+                cancellationConfirmed = true;
+                modal.close();
+                e.currentTarget.click();
+              });
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              $('[data-cancel-dialog]', modal.body).click((event) => {
+                modal.close();
+              });
+            },
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            no_confirmation_needed(modal, jsonData) {
+              modal.close();
+              cancellationConfirmed = true;
+              e.currentTarget.click();
+            },
+          },
+          triggerElement: e.currentTarget,
```

Comment:
Make sure this line isn't missing when we move the code from the inline script to here, as this is part of the fix in #11435